src/components/FilterSingleSelect/FilterSingleSelect.tsx

```diff
@@ -17,6 +17,8 @@ export default function FilterSingleSelect(props: FilterSingleSelectProps): Reac
 		options: []
 	});
 
+	if (props.required && !props.data.value) throw new Error("A value is required")

```

Comment:
I'd suggest adding a more complete error message like "Invalid use-case, a value is required but none was provided" so developers know where and how to fix their error.